### PR TITLE
http-gateway: expose react app

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: bundle/Dockerfile [hunk 1]
```diff
@@ -48,6 +48,14 @@ RUN mkdir -p ./nats-server
 RUN unzip ./nats-server.zip -d ./nats-server
 RUN cp ./nats-server/*/nats-server /go/bin/nats-server
 
+FROM ubuntu:20.04 AS build-web
```
Comment (ondrejtomcik) [inline bundle/Dockerfile]: Why you are not using node container?

File: bundle/run.sh [hunk 2]
```diff
@@ -247,6 +247,7 @@ Dial:
 JwksURL: ${JWKS_URL}
 ResourceDirectoryAddr: ${RESOURCE_DIRECTORY_ADDRESS}
 CertificateAuthorityAddr: ${CERTIFICATE_AUTHORITY_ADDRESS}
+WWWDirectory: /usr/local/var/www
```
Comment (ondrejtomcik) [inline bundle/run.sh]: This should be the default value in the http-gw service.

File: http-gateway/Dockerfile [hunk 3]
```diff
@@ -2,7 +2,16 @@ FROM cloud-build AS build
 WORKDIR $GOPATH/src/github.com/plgd-dev/cloud/http-gateway
 RUN go build -o /go/bin/service ./cmd/service
 
+FROM ubuntu:20.04 AS build-web
```
Comment (ondrejtomcik) [inline http-gateway/Dockerfile]: Use node container.

File: http-gateway/service/httpApi.go [hunk 4]
```diff
@@ -120,6 +117,11 @@ func NewHTTP(requestHandler *RequestHandler, authInterceptor kitHttp.Interceptor
 	r.HandleFunc(uri.WsStartDevicesObservation, requestHandler.startDevicesObservation).Methods(http.MethodGet)
 	r.HandleFunc(uri.WsStartDeviceResourcesObservation, requestHandler.startDeviceResourcesObservation).Methods(http.MethodGet)
 
+	// serve www directory
+	if requestHandler.config.WWWDirectory != "" {
```
Comment (ondrejtomcik) [inline http-gateway/service/httpApi.go]: UI will be always embedded in the http gateway container. It's more clean to use boolean value to set if the UI should be hosted or not.

**Review by ondrejtomcik — CHANGES_REQUESTED** [review]

File: bundle/Dockerfile (hunk 1 above)
Comment (jkralik) [reply bundle/Dockerfile]: Fixed

File: bundle/run.sh (hunk 2 above)
Comment (jkralik) [reply bundle/run.sh]: Fixed

File: http-gateway/Dockerfile (hunk 3 above)
Comment (jkralik) [reply http-gateway/Dockerfile]: Fixed

File: http-gateway/service/config.go [hunk 5]
```diff
@@ -9,14 +9,16 @@ import (
...
+	Address                  string
+	Listen                   certManager.Config
+	Dial                     certManager.Config
+	JwksURL                  string
+	ResourceDirectoryAddr    string
+	CertificateAuthorityAddr string
+	WebSocketReadLimit       int64
+	WebSocketReadTimeout     time.Duration
+	UIEnabled                bool
+	UIDirectory              string
```
Comment (ondrejtomcik) [inline http-gateway/service/config.go:21]: What is the use case to have it configurable ?

Comment (jkralik) [reply http-gateway/service/config.go:21]: I don't want to hard-code it, but I don't have use case for now. So I want to leave as is.

**Review by ondrejtomcik — APPROVED** [review]

**Pull request merged**